pkg/utils/nodes.go

```diff
@@ -0,0 +1,24 @@
+package utils
+
+import (
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// CheckNodeName returns an error if nodeName doesn't match any node name in the cluster, otherwise a nil
+func CheckNodeName(r client.Reader, nodeName string) error {
+	node := &corev1.Node{}
+	objNodeName := types.NamespacedName{Name: nodeName}
+	err := r.Get(context.Background(), objNodeName, node)
+	if err != nil {
+		// no match between CR name and the cluster nodes
+		msg := fmt.Sprintf("node %s is invalid", nodeName)
+		err = buildApiError(err, msg)
+		return fmt.Errorf("CR's name doesn't match any node name in the cluster - %w", err)
```

Comment:
I think it would simply things to replace this by wrapping the error.
Something like:
```
"github.com/pkg/errors"
...
errors.Wrapf(err, "node %s is invalid", nodeName)
```

Later on in `Reconcile` when you need to check whether node was found or not you can use the error cause to do that:
```
if apiErrors.IsNotFound(errors.Cause(err)) {
	return emptyResult, err
}
```

Reply:
> Later on in Reconcile...

That check is better placed here IMHO, because isNotFound is a valid result for this function and not an error.

Reply:
>  isNotFound is a valid result for this function and not an error.

> node found = name valid
node not found = name invalid
any other error = retry

When `node not found = name invalid`, then we should not return an error _CheckNodeName_ function and in reconcile or are you referring that _CheckNodeName_ function should not return an error but in reconcile we should raise an error (since we didn't found a match between CR's name and cluster node names)?

I am still not sure what is your suggestion. I went with the second option.

Reply:
> the func signature might need an update as well (atm the is prefix suggests that the return value is a boolean, bit it's an error)

you removed the `is` prefix... what about returning a bool instead? That would make things easier...

```
func Reconcile() {
  ...
  isValid, err := isNodeNameValid(name)
  if err != nil {
    return err
  } else if !isValid {
    // log, event, whatever, and return without error, because a requeue does not make sense
    return result, nil
  {
  // remediate...
}

func isNodeNameValid(name) (bool, error) {
  err := getNode(name)
  if isNotFound(err) {
    return false, nil
  } else if err != nil {
    return false, err // bool value doesn't matter
  }
  return true, nil
}
```